rust-src/id/src/id_verifier.rs

```diff
@@ -0,0 +1,173 @@
+        keys,
+    )
+    .is_ok()
+}
+
+/// Function for verifying account ownership. The arguments are
+/// - public_data - the public keys (and threshold) of the prover. These should
+///   be read from chain by looking up the account. If they are not present on
+///   chain, the verifier should reject the proof.
+/// - account - the account address of the account that the prover claims to
+///   own.
+/// - challenge - the challenge that the verifier gave to the prover.
+/// - proof - the prover's proof.
+///
+/// The function outputs a bool, indicating whether the proof is correct or not,
+/// i.e., it checks the signatures inside the proof.
+pub fn verify_account_ownership(
+    public_data: &CredentialPublicKeys,
+    account: AccountAddress,
+    challenge: &[u8],
+    proof: &AccountOwnershipProof,
+) -> bool {
+    let mut hasher = Sha256::new();
+    hasher.update(account.0);
+    hasher.update([0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8]);
+    hasher.update(b"account_ownership_proof");
+    hasher.update(&challenge);
+    let to_sign = &hasher.finalize();
+    utils::verify_account_ownership_proof(&public_data.keys, public_data.threshold, &proof, to_sign)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{constants::AttributeKind, id_prover::*};
+    use crypto_common::types::{KeyIndex, KeyPair};
+    use pairing::bls12_381::G1;
+    use rand::*;
+    use std::collections::btree_map::BTreeMap;
+
+    #[test]
+    fn test_verify_account_ownership() {
+        let mut csprng = thread_rng();
+
+        let cred_data = CredentialData {
+            keys:      {
+                let mut keys = BTreeMap::new();
+                keys.insert(KeyIndex(0), KeyPair::generate(&mut csprng));
+                keys.insert(KeyIndex(1), KeyPair::generate(&mut csprng));
+                keys.insert(KeyIndex(2), KeyPair::generate(&mut csprng));
+                keys
+            },
+            threshold: SignatureThreshold(2),
+        };
+
+        let pub_data = cred_data.get_cred_key_info();
+
+        let reg_id: G1 = Curve::hash_to_group(b"some_bytes");
+        let account_address = AccountAddress::new(&reg_id);
+        let challenge = b"13549686546546546854651357687354";
+
+        let proof = prove_ownership_of_account(cred_data, account_address, challenge);
+
+        assert!(verify_account_ownership(
+            &pub_data,
+            account_address,
+            challenge,
+            &proof
+        ));
+    }
+
+    #[test]
+    fn test_verify_attribute() {
+        let mut csprng = thread_rng();
+        let keys = PedersenKey::<G1>::generate(&mut csprng);
+        let attribute = AttributeKind("some attribute value".to_string());
+        let value = Value::<G1>::new(attribute.to_field_element());
+        let (commitment, randomness) = keys.commit(&value, &mut csprng);
+
+        assert!(verify_attribute(
+            &keys,
+            &attribute,
+            &randomness,
+            &commitment
+        ));
```

Comment:
Recommend adding a short string description to the assertion so that if the test fails it is easier to find and understand what the intention was, e.g.,